ci/test_cpp.sh

```diff
@@ -8,9 +8,10 @@ trap "EXITCODE=1" ERR
 set +e
 
 # Get library for finding incorrect default stream usage.
-STREAM_IDENTIFY_LIB="${CONDA_PREFIX}/lib/libcudf_identify_stream_usage.so"
+STREAM_IDENTIFY_LIB_MODE_CUDF="${CONDA_PREFIX}/lib/libcudf_identify_stream_usage_mode_cudf.so"
+STREAM_IDENTIFY_LIB_MODE_TESTING="${CONDA_PREFIX}/lib/libcudf_identify_stream_usage_mode_testing.so"
```

Comment:
Not in this PR, but it will be used in immediate follow-ups like #12090 and it's conceptually easier to explain why it's necessary as part of this PR because this PR is where the library has been modified to be compiled twice.